mflike/mflike.py

```diff
@@ -425,10 +425,17 @@ def get_sacc_names(pol, exp_1, exp_2):
         self.logp_const -= 0.5 * np.linalg.slogdet(self.cov)[1]
 
         self.experiments = data["experiments"]
-        self.bands = {
-            name: {"nu": tracer.nu, "bandpass": tracer.bandpass}
-            for name, tracer in s.tracers.items()
-        }
+        self.bands = {}
+        self.beams = {}
+        for name, tracer in s.tracers.items():
+            self.bands[name] = {"nu": tracer.nu, "bandpass": tracer.bandpass}
+            # trying to read beams, if present
+            try:
+                tracer.beams
+            except:
+                pass
+            else:
+                self.beams[name] = {"nu": tracer.nu, "beams": tracer.beam}
```

Comment:
Here I'm not sure I get the logic since you try to get the attribute `beams` of tracer but then you fill the `self.beams` dict with `tracer.beam`. I think you would like to fill the `self.beams` dict in case the attribute `beam` of tracer exists right ? If yes, then the `beam` attribute of tracer always exists so I'm not sure you need the `try/except`. In any case, it's better to avoid `try/except` block and you can can test the existence of class atrribute with `hasattr` function so
```python 
if hasattr(tracer, "beam"):
  self.beams[name] = {"nu": tracer.nu, "beams": tracer.beam}
```

Reply:
Hi @xgarrido , yes, the `tracer.beams` in 
`try:`
 `               tracer.beams`
is a typo. And thanks for your suggestion, will apply that instead.

Anyway I just put that piece of code as a placeholder, since so far `tracer.beam` can only be an array of dimension (ells), while I would like it to be of dimensions (freqs, ells)